test/cql-pytest/test_aggregate.py

```diff
@@ -72,17 +72,24 @@ def test_count_in_partition(cql, table1):
     cql.execute(stmt, [p, 3, 3])
     assert [(3,)] == list(cql.execute(f"select count(*) from {table1} where p = {p}"))
 
-# Using count(v) instead of count(*) allows counting only rows with a set
+# Using count(v) instead of count(*) allows counting only rows with a non-NULL
 # value in v
-def test_count_specific_column(cql, table1):
+def test_count_specific_column(cql, test_keyspace, table1):
     p = unique_key_int()
     stmt = cql.prepare(f"insert into {table1} (p, c, v) values (?, ?, ?)")
     cql.execute(stmt, [p, 1, 1])
     cql.execute(stmt, [p, 2, 2])
-    cql.execute(stmt, [p, 3, 3])
+    cql.execute(stmt, [p, 3, 3,])
     cql.execute(stmt, [p, 4, None])
     assert [(4,)] == list(cql.execute(f"select count(*) from {table1} where p = {p}"))
     assert [(3,)] == list(cql.execute(f"select count(v) from {table1} where p = {p}"))
+    # Check with non-scalar types too, reproduces #14198
+    cql.execute(f'CREATE TYPE {test_keyspace}.my_udt (i int, t text)');
+    cql.execute(f'ALTER TABLE {table1} ADD (fli frozen<list<int>>, tup tuple<int, bigint>, udt {test_keyspace}.my_udt)')
```

Comment:
The fixture "table1" can be shared between multiple tests, when you use it you need to think about that, and in particular it's a mistake to ALTER TABLE it. What you did in your first patch - modify the table1 definition to add more columns - is better. Alternatively, you can just create a table2 fixture.

Reply:
But how can I use new_type with a new table2 fixure?